File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/adapter/NavigationHeaderHolderFactory.java
package io.github.romangolovanov.apps.ametro.ui.navigation.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import io.github.romangolovanov.apps.ametro.R;

class NavigationHeaderHolderFactory implements IHolderFactory{

    @Override
    public IHolder createHolder(View convertView) {
        NavigationHeaderHolder holder = new NavigationHeaderHolder(convertView);
        convertView.setTag(holder);
        return holder;
    }

    @Override
    public View createView(LayoutInflater inflater, ViewGroup parent) {
        return inflater.inflate(R.layout.drawer_header_item, parent, false);
    }

}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/navigation/adapter/NavigationHeaderHolderFactory.kt
package io.github.romangolovanov.apps.ametro.ui.navigation.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import io.github.romangolovanov.apps.ametro.R

internal class NavigationHeaderHolderFactory : IHolderFactory {

    override fun createHolder(view: View): IHolder {
        val holder = NavigationHeaderHolder(view)
        view.tag = holder
        return holder
    }

    override fun createView(inflater: LayoutInflater, parent: ViewGroup): View {
        return inflater.inflate(R.layout.drawer_header_item, parent, false)
    }
}
